src/directives/add-or-remove-lang-cy.directive.ts

```diff
@@ -6,21 +6,27 @@ import { TranslateService } from '@ngx-translate/core';
 })
 export class AddOrRemoveLangCyDirective {
 
-  constructor (
+  constructor(
     private ref: ElementRef,
     private translateService: TranslateService,
     private renderer: Renderer2,
-  ) {}
+  ) {
+  }
 
-  ngOnInit() {
-    this.translateService.store.onLangChange.subscribe( (value) => {
-      if (value.lang) {
-        if (value.lang === 'cy') {
-          this.renderer.setAttribute(this.ref.nativeElement, 'lang', 'cy');
-        } else {
-          this.renderer.removeAttribute(this.ref.nativeElement, 'lang');
-        }
+  assignLanguage(language: string) {
+    if (language) {
+      if (language === 'cy') {
+        this.renderer.setAttribute(this.ref.nativeElement, 'lang', 'cy');
+      } else {
+        this.renderer.removeAttribute(this.ref.nativeElement, 'lang');
       }
+    }
+  }
+
+  ngOnInit() {
+    this.assignLanguage(this.translateService.store.currentLang);
+    this.translateService.store.onLangChange.subscribe((value) => {
+      this.assignLanguage(value.lang);
```

Comment:
do we need to set the value first, then subscribe to it, then set it again? seems backwards